apps/crud/src/app/components/todo-item/todo-item.component.spec.ts

```diff
@@ -0,0 +1,88 @@
+import { fireEvent, render, screen } from '@testing-library/angular';
+import { TodoItemComponent } from './todo-item.component';
+import { provideComponentStore } from '@ngrx/component-store';
+import { TodoItemStore } from '../../todoItem.store';
+import { HttpClient, HttpHandler } from '@angular/common/http';
+import { AppStore } from '../../app.store';
+import { TodoService } from '../../service/todo.service';
+import { LoadingSpinnerComponent } from '../loading-spinner/loading-spinner.component';
+import { createMockWithValues } from '@testing-library/angular/jest-utils';
+import { TestBed } from '@angular/core/testing';
+import { of } from 'rxjs';
+
+describe('TodoItemComponent', () => {
+  test('should render', async () => {
+    const { fixture } = await setup();
+    expect(fixture).toBeTruthy();
+  });
+
+  test('update', async () => {
+    const { mockTodoService, fixture } = await setup();
+
+    fixture.detectChanges();
+
+    const button = screen.getAllByRole('button')[0];
+
+    fireEvent.click(button);
+
+    expect(mockTodoService.updateTodo).toHaveBeenCalledTimes(1);
+  });
+
+  test('delete', async () => {
+    const { mockTodoService, fixture } = await setup();
+
+    fixture.detectChanges();
+
+    const button = screen.getAllByRole('button')[1];
```

Comment:
getting all buttons is dangerous, because if you add another buttons, this can break the test and create a false positive. 

Prefer `screen.getByRole('button', {name: 'xxx'})`